pdns/dnsdist.cc

```diff
@@ -638,7 +638,12 @@ try {
 #ifdef HAVE_DNS_OVER_HTTPS
             // DoH query
             du->response = std::string(response, responseLen);
-            if (send(du->rsock, &du, sizeof(du), 0) != sizeof(du)) {
+            ssize_t sent = write(du->rsock, &du, sizeof(du));
+            if (sent != sizeof(du)) {
+              if (errno == EAGAIN || errno == EWOULDBLOCK) {
+                ++g_stats.dohResponsePipeFull;
+              }
+
```

Comment:
maybe internal erorr on other errors?

Reply:
I don't really know what to do in that case besides releasing the memory.. We can't trigger a response from libh2o because we are not in the right thread for that, and we can't talk to that thread without writing to this pipe. So apart from releasing the memory.. Perhaps I could log an error in verbose mode?